onnxruntime/core/graph/ep_graph_utils.cc

```
@@ -0,0 +1,115 @@
+        graph_utils::MakeInitializerCopyIfNotExist(graph_viewer.GetGraph(), new_graph, input->Name(),
+                                                   copy_in_memory_initializer);
+      }
+    }
+
+    // Updates node attributes if any.
+    // Ex: if the node has subgraph, it's possible that the subgraph and the GraphProto in node attribute are not in sync because of graph optimization.
+    // Therefore, we need to force GraphProto attribute to be updated in order to get the valid GraphProto.
+    if (node.GetAttributes().size() > 0) {
+      auto node_proto = std::make_unique<ONNX_NAMESPACE::NodeProto>();
+      // we need to update any GraphProto attributes for subgraphs so that any changes made by things
+      // such as the optimizers are captured. otherwise we can end up saving an invalid graph.
+      node.ToProto(*node_proto, true);  // update subgraphs
+      const int num_attributes = node_proto->attribute_size();
+      auto node_attributes = std::make_unique<NodeAttributes>();
+      node_attributes->reserve(num_attributes);
+
+      for (int i = 0; i < num_attributes; ++i) {
+        auto& attr = node_proto->attribute(i);
+        node_attributes->emplace(attr.name(), attr);
+      }
+
+      // The GraphProto attributes are the updated ones.
+      new_graph.AddNode(node.Name(), node.OpType(), node.Description(), inputs, outputs, node_attributes.get(), node.Domain());
+    } else {
+      // The GraphProto attributes are the original ones.
+      new_graph.AddNode(node.Name(), node.OpType(), node.Description(), inputs, outputs, &node.GetAttributes(), node.Domain());
+    }
+  }
+
+  // TODO: There is an edge case where the OrtValueInfo is outer scope and it's upper-level graph's input (not the initializer).
+
+  ORT_API_RETURN_IF_STATUS_NOT_OK(new_graph.Resolve());
+  out_model = std::move(model);
+#else
+  fprintf(stderr, "The GetSubGraphAsModelFromGraph is not supported in a minimal build.\n");
```

Should this return an error `Status` instead of printing?